vm_supervisor/pool.py

```diff
@@ -47,7 +47,7 @@ async def get_running_vm(self, vm_hash: VmHash) -> Optional[VmExecution]:
             return None
 
     def forget_vm(self, vm_hash: VmHash) -> None:
-        self.executions.pop(vm_hash)
+        self.executions.pop(vm_hash, None)
```

Comment:
```
try:
    del executions[vm_hash]
except KeyError:
    pass
```

would convey a better meaning IMO since you don't use the returned value, but your solution works as well.

Reply:
More obvious, sounds good.